core/impl/src/main/java/com/blazebit/persistence/impl/AbstractCommonQueryBuilder.java

```diff
@@ -1829,7 +1829,7 @@ protected String getBaseQueryStringWithCheck() {
         for (JoinNode node : joinManager.getEntityFunctionNodes()) {
             Class<?> clazz = node.getJavaType();
             int valueCount = node.getValueCount();
-            boolean identifiableReference = node.getValuesIdName() != null;
+            boolean identifiableReference = node.getValuesIdNames() != null && !node.getValuesIdNames().isEmpty();
```

Comment:
Should `null` default to `Collections.emptySet()`?

Reply:
Sounds reasonable.